### PR TITLE
added forwardRef + imperativa handle to handle inner methods

### PR DESCRIPTION
**Pull request by matheusnasser**
Implemented forwardRef + imperative Handle to enable accessing component methods from outside, allowing users to control the component externally.

Introduced the slideOnClick prop, which enables users to navigate directly to a specific item in the carousel by clicking on it.

I would greatly appreciate your thoughts on how to enhance the code and incorporate additional features.

Thank you in advance,
Matheus Nasser

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -14,9 +15,19 @@ export type CarouselProps = {
...
 	readonly showControls?: boolean;
+	readonly slideOnClick?: boolean;
+	readonly ref?: React.ForwardedRef<CarouselRef>;
 };
 
-export const Carousel: React.FC<CarouselProps> = (props: CarouselProps) => {
+export type CarouselRef = {
+	readonly next: () => void;
+	readonly prev: () => void;
+	getItems: () => CarouselItem[];
```
Comment (scriptex) [inline src/index.tsx:25]: Please make these properties `readonly`. You can also use the `Readonly<T>` if you don't want to explicitly write it for each prop.

File: src/index.tsx [hunk 2]
```diff
@@ -28,7 +39,7 @@ export const Carousel: React.FC<CarouselProps> = (props: CarouselProps) => {
 	const getSlideStyle = (index: number): React.CSSProperties => {
 		const style: React.CSSProperties = {};
 
-		if (index < len) {
+		if (index <= len) {
```
Comment (scriptex) [inline src/index.tsx:42]: Why do we need this change here?

File: src/index.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ export type CarouselItem = {
 	readonly alt?: string;
 	readonly content: React.ReactNode;
 	readonly image: string;
+	readonly click?: () => void;
```
Comment (scriptex) [inline src/index.tsx:8]: Can we please rename this to `onClick` to indicate that it is actually an action?

**Review by scriptex — CHANGES_REQUESTED** [review]
Hey, @matheusnasser, thank you very much for your contribution!
I've added some minor comments and questions, can you please check them out? Thanks! ❤️

File: src/index.tsx (hunk 3 above)
Comment (matheusnasser) [reply src/index.tsx:8]: surething

**Comment by scriptex** [comment]
Hey, @matheusnasser, can you please update the demo by adding the new configuration settings?

**Comment by matheusnasser** [comment]
I'll work on that.

**Comment by scriptex** [comment]
@matheusnasser if the demo causes troubles we can try and merge your code and then I take over updating the demo.

**Comment by matheusnasser** [comment]
I've been busy with some stuff at work that's why i wasn't able to update it, sry for that.

If you want to take over the demo would be awesome.

**Comment by scriptex** [comment]
@matheusnasser the demo has been updated and the new functionality has been released to NPM. Please download and install v1.4.0
Thank you very much for your contribution! ❤️

**Comment by matheusnasser** [comment]
Thank you for the support and updating the demo! rly appreciate it! Happy to help!

**Pull request merged**